src/main/java/com/simple/autocomplete/lucene/service/SearchService.java

```diff
@@ -0,0 +1,128 @@
+package com.simple.autocomplete.lucene.service;
+
+import com.simple.autocomplete.lucene.analyzer.CustomNgramAnalyzer;
+import com.simple.autocomplete.title.domain.TitleInfo;
+import com.simple.autocomplete.utils.Eng2KorConverter;
+import com.simple.autocomplete.utils.PropertyLoader;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queries.function.FunctionScoreQuery;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.*;
+import org.apache.lucene.store.MMapDirectory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * @author LEE
+ * @contact ljh0326s@gmail.com
+ */
+public class SearchService {
+    //색인 정보가 저장된 경로
+    private final Logger LOGGER = LoggerFactory.getLogger(SearchService.class);
+
+    /**
+     * 검색 결과를 반환하는 메서드
+     * @param searchValue 검색어
+     * @return 검색결과
+     * @throws IOException
+     */
+    public Set<TitleInfo> search(String searchValue) throws IOException {
+        return search(searchValue, 10, false);
+    }
+
+    public Set<TitleInfo> search(String searchValue, int searchCount) throws IOException {
+        return search(searchValue, searchCount, false);
+    }
+
+    public Set<TitleInfo> search(String searchValue, int searchCount, boolean showScoreLog) throws IOException {
+
+        Set<TitleInfo> result = new LinkedHashSet<>();
+        //영 -> 한 변환기 ex) 안녕 -> dkssud
+        Eng2KorConverter eng2Kor = new Eng2KorConverter();
+        //인댁스 경로. property.properties 참조
+        String indexPath = PropertyLoader.getInstance().getPropertyValue("INDEX_DIR_PATH");
+
+        //빈 글자라면 검색하지말고 결과 반환
+        if(searchValue.isEmpty()) return result;
```

Comment:
```suggestion
        if(searchValue.isEmpty()) {
          return result;
        }
```

LOC가 1줄에서 3줄 늘어나긴 하지만 가독성 측면에서 중괄호를 넣는 편이 좋은것 같습니다.

Reply:
옙 이게 더 좋은것같아요!!!